src/scripts/main.js

```diff
@@ -1,3 +1,373 @@
+
+      if (ageCell || salaryCell) {
+        input.type = 'number';
+      }
+
+      input.className = 'cell-input';
+      input.value = tdText;
+      td.innerHTML = '';
+      td.append(input);
+      input.focus();
+      savedText(input);
+    }
+
+    function savedText(tag) {
+      tag.addEventListener('blur', () => {
+        deleteNotification();
+        replaceValue(tag);
+      });
+
+      table.addEventListener('keypress', (e) => {
+        if (e.code === 'Enter') {
+          deleteNotification();
+          replaceValue(tag);
+        }
+      });
+    }
+
+    function replaceValue(tag) {
+      const emptyValue = tag.value.length === 0;
+
+      if (emptyValue) {
+        td.textContent = tdText;
+
+        pushNotification(450, 190, 'Error',
+          'The cell must not be empty!!! '
+          + 'The data are not changed!!!',
+          'error');
+
+        return;
+      }
+
+      switch (true) {
+        case nameCell:
+          if (tag.value.length < minNameChars) {
+            td.textContent = tdText;
+
+            pushNotification(450, 190, 'Error',
+              `The name must contain at least ${minNameChars} letters`,
+              'error');
+          } else {
+            pushNotification(450, 190, 'Success',
+              'The person\'s age has been successfully changed', 'success');
+            td.textContent = tag.value;
+          }
+          break;
+
+        case positionCell:
+          if (tag.value.length < minPositionChars) {
+            td.textContent = tdText;
+
+            pushNotification(450, 190, 'Error',
+              'The name of the position must be'
+              + `contain at least ${minPositionChars} letters`,
+              'error');
+          } else {
+            pushNotification(450, 190, 'Success',
+              'The person\'s position has been successfully changed',
+              'success');
+            td.textContent = `${tag.value}`;
+          }
+          break;
+
+        case ageCell:
+          if (tag.value < minYear || tag.value > maxYear) {
+            td.textContent = tdText;
+
+            pushNotification(450, 190, 'Error',
+              `Age must be over ${minYear} and no more than ${maxYear}`,
+              'error');
+          } else {
+            pushNotification(450, 190, 'Success',
+              'the person\'s age has been successfully changed', 'success');
+            td.textContent = `${tag.value}`;
+          }
+          break;
+
+        case salaryCell:
+          if (tag.value < minSalary) {
+            td.textContent = tdText;
+
+            pushNotification(450, 190, 'Error',
+              `Salary cannot be less than ${minSalary}`,
+              'error');
+          } else {
+            pushNotification(450, 190, 'Success',
+              'a person\'s salary has been successfully changed', 'success');
+            td.textContent = `$${parseInt(tag.value).toLocaleString('en-EN')}`;
```

Comment:
you can also add `{style:"currency", currency:"USD"}` option for `toLocaleString`

Reply:
https://www.w3schools.com/jsref/tryit.asp?filename=tryjsref_tolocalestring_num_options3